tinygrad/codegen/uops.py

```diff
@@ -77,30 +77,19 @@ class UPat:
   dtype: Optional[Union[DType, Set[DType]]] = None
   allow_len: Set[int] = field(default_factory=set)
 
-  @classmethod
-  def from_dict(cls, pat:Dict[str, Any]) -> UPat:
-    name, uop, dtype = pat.get("__name__"), pat.get("uop"), pat.get("dtype")
-    assert isinstance(name, str) or name is None
-    assert isinstance(uop, (UOps, set)) or uop is None
-    assert isinstance(dtype, (DType, set)) or dtype is None
-    vin = pat.get("vin")
-    if isinstance(vin, list): vin = [UPat.from_dict(x) for x in vin]
-    elif isinstance(vin, tuple): vin = tuple(UPat.from_dict(x) for x in vin)
-    elif isinstance(vin, dict): vin = UPat.from_dict(vin)
-    else: assert vin is None
-    arg = pat.get("arg")
-    allow_len = pat.get("__allow_len__", set())
-    return cls(uop, arg, vin, name, dtype, allow_len)
-
 def _match(uop:UOp, pat:UPat, store:Dict[str, UOp]) -> bool:
-  if pat.name in store and store[pat.name] != uop: return False
-  if pat.name is not None: store[pat.name] = uop
-  if isinstance(pat.arg, set) and uop.arg not in pat.arg: return False
-  elif pat.arg is not None and uop.arg != pat.arg: return False
-  if isinstance(pat.dtype, set) and uop.dtype not in pat.dtype: return False
-  if isinstance(pat.dtype, DType) and uop.dtype != pat.dtype: return False
-  if isinstance(pat.uop, set) and uop.uop not in pat.uop: return False
-  if isinstance(pat.uop, UOps) and uop.uop != pat.uop: return False
+  if pat.name is not None:
+    if pat.name in store and store[pat.name] != uop: return False
+    store[pat.name] = uop
+  if pat.arg is not None and uop.arg != pat.arg: return False
```

Comment:
Err, arg can be a set now after the PTX shift addition? Why was this not caught by process replay?

Reply:
Oops, must've accidentally skipped over this when rebasing, will also add a regression test for this

Reply:
fwiw the previous implementation did not pass the regression test i just added